src/viewer.js

```diff
@@ -3445,16 +3502,47 @@ function onRotateLeft() {
 function onRotateRight() {
     if ( this.viewport ) {
         var currRotation = this.viewport.getRotation();
-        if (currRotation === 270) {
-            currRotation = 0;
-        }
-        else {
-            currRotation += 90;
+        if ( this.flipped ){
+          if (currRotation === 0) {
+              currRotation = 270;
+          }
+          else {
+              currRotation -= 90;
+          }
+        } else {
+          if (currRotation === 270) {
+              currRotation = 0;
+          }
+          else {
+              currRotation += 90;
+          }
         }
         this.viewport.setRotation(currRotation);
     }
 }
 
+/**
+ * Note: When pressed f on keyboard or flip control button
+ */
+function onButtonFlip() {
```

Comment:
I see... I didn't really realize that the two functions are called from different contexts. I guess it's not too big of a deal; it just bothers me that they are identical functions otherwise. I think I'd make `onKeyboardFlip` take in a `viewer` argument (and in the keyboard handler you could call it with `onKeyboardFlip(this.viewer)`) and you could replace all of the code in `onButtonFlip` with just `onKeyboardFlip(this)`. That way we're not duplicating the code (which can lead to maintenance issues in the future). Of course with that change it would maybe make more sense to change the name of `onKeyboardFlip` back to `onFlip` or something. Apologies for the back and forth on this issue. :)

Reply:
No problem :) always learning

Reply:
Done! The flip state variable is now stored on viewport objects too :)